pact4s/src/test/scala/org/broadinstitute/dsde/workbench/sam/provider/SamProviderSpec.scala

```diff
@@ -209,12 +227,16 @@ class SamProviderSpec
     // how to verify external states of cloud services through mocking and stubbing
     .withStateManagementFunction(
       StateManagementFunction {
```

Comment:
nit - there's quite a bit to read here and it can get confusing when there are different layers of nesting.  I would prefer that the inline object instantiations be extracted into a variable that you can then call pass to the `.withX` methods on the builder.

Reply:
Lifted the inline object creation `StateManagementFunction` into a variable called `providerStatesHandler`.